Automattic/wp-calypso
# Running wp-e2e-tests in wp-calypso

#### Changes proposed in this Pull Request

This PR pulls in all of the e2e tests from wp-e2e-tests. The CircleCI config has been updated to run Canary tests ( IE, Safari, and WooCommerce) as well as the full suite of e2e tests. 

More updates will come in future PRs. These include:
* Clean up documentation
* Clean up code so some eslint overrides can be removed
* Removing unneeded scripts

p6fDka-Bw-p2

#### Testing instructions

First, Ensure that tests run and pass in CircleCI

Test them locally:
* run `npm run test-e2e` and make sure tests start to run. Chrome instances should start opening and performing actions. 

Run a single spec file:
* run `cd test/e2e`
* run `./node_modules/.bin/mocha specs/wp-page-editor-spec.js`

Most of the documentation in the README with regards to running tests is still accurate as long as you are in the `test/e2e` directory

It isn't necessary that all of the tests run and pass locally as this can take a while and be resource intensive, but as long as they start and pass in CircleCI, it is good.
